packages/react/src/hooks/useStorageURL.ts

```diff
@@ -14,31 +14,35 @@ export interface UseStorageURLResult {
  */
 export const useStorageURL = (
   key: string,
-  options?: S3ProviderGetConfig
+  options?: S3ProviderGetConfig,
+  fallbackURL?: string
 ): UseStorageURLResult & { fetch: () => () => void } => {
   const [result, setResult] = React.useState<UseStorageURLResult>({
     isLoading: true,
   });
 
   // Used to prevent an infinite loop on useEffect, because `options`
   // will have a different reference on every render
-  const serializedOptions = JSON.stringify(options);
+  const optionsRef = React.useRef(options);
 
   const fetch = () => {
     setResult({ isLoading: true });
 
+    const options = optionsRef.current;
     const promise = Storage.get(key, options);
 
     // Attempt to fetch storage object url
     promise
       .then((url) => setResult({ url, isLoading: false }))
-      .catch((error: Error) => setResult({ error, isLoading: false }));
+      .catch((error: Error) =>
+        setResult({ url: fallbackURL, error, isLoading: false })
+      );
 
     // Cancel current promise on unmount
     return () => Storage.cancel(promise);
   };
 
-  React.useEffect(fetch, [key, options, serializedOptions]);
+  React.useEffect(fetch, [key, fallbackURL]);
```

Comment:
The usage of `key` and `fallbackUrl` as dependencies of this effect is opaque, can you please add a comment on why `fetch` should be rerun if they change? Actually, should `fetch` rerun if `fallbackUrl` changes?

Reply:
Sure, will add a comment. I was thinking initialize url to `fallbackURL` so that we do not need to put it into `useEffect` but then realized this was wrong because that way `img` will try to load the fallback in the first place 😅 .

Reply:
Still unclear on why a change to `fallbackUrl` should trigger a new fetch?

Reply:
Think we can move it out of the `useEffect` by:
```js
...
if (result.error) {
   result.url = fallbackURL;
}
...
```